modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -645,9 +646,44 @@ private boolean modelIsMutatable(CodegenModel model, Set<String> processed) {
         return isMutatable;
     }
 
-    protected void removePropertiesDeclaredInComposedTypes(Map<String, ModelsMap> objs, CodegenModel model, List<CodegenProperty> composedProperties) {
+    protected final Stream<CodegenModel> getSubModels(Map<String, ModelsMap> models, CodegenModel model)
```

Comment:
Please add docs for this method. Also, we may want to consider another name.

Reply:
The concept is called a `subschema` in [swagger.io](https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/). I have chosen for `submodel` to stay more in-line with the terminology in this codebase. 

Do you have a suggestion for a better name?

Reply:
Since it doesn't include allOf I would think it might be named getOneOfAndAnyOfSubModels, but I'm not too pressed about what you use.

Reply:
I got your point. But the implementation filters on references as well. So then it should be `getOneOfAndAnyOfReferrencedModels`. But it becomes a very technical instead of functional name.

Why is `allOf` actually excluded? And why are in-line schemas excluded?
The answer might give a hint for a proper method name.
Do you have a clue?

Reply:
I think allOf doesn't need a discriminator.

Reply:
hm.. according to [the spec](https://spec.openapis.org/oas/v3.1.0#discriminatorObject) it can:
> The discriminator object is legal only when using one of the composite keywords `oneOf`, `anyOf`, `allOf`.

I tried to understand the `DefaultCodegen.fromModel()` code flow to review the behavior of `allOf`, but its too complex.
I propose to keep the name as is for now.